lib/neo4j/shared/type_converters.rb

```diff
@@ -35,55 +41,31 @@ def db_type
         end
 
         def to_db(value)
-          value.to_i
+          val = value.to_i
+          val > NEO4J_LARGEST_INT ? val.to_s : val
         end
 
-        alias to_ruby to_db
-      end
-    end
-
-    class FloatConverter < BaseConverter
-      class << self
-        def convert_type
-          Float
-        end
-
-        def db_type
-          Float
-        end
-
-        def to_db(value)
-          value.to_f
+        def to_ruby(value)
+          value.to_i
         end
-        alias to_ruby to_db
       end
     end
 
-    class BigDecimalConverter < BaseConverter
-      class << self
-        def convert_type
-          BigDecimal
-        end
-
-        def db_type
-          String
-        end
-
-        def to_db(value)
-          case value
-          when Rational
```

Comment:
should we handle `Rational` under `Float`

Reply:
Since `Rational` has `.to_f` method, those will be supported.